AstSemantics.md

```diff
@@ -248,6 +295,32 @@ The details of index space for local variables and their types will be further c
 e.g. whether locals with type `i32` and `i64` must be contiguous and separate from
 others, etc.
 
+## Global variables
+
+A *global variable* stores a single value of a fixed [value type](#types) and may be
+declared either *mutable* or *immutable*. This provides WebAssembly with memory
+locations that are disjoint from any [linear memory](#linear-memory) and thus
+cannot be arbitrarily aliased as bits.
+
```

Comment:
For immutable globals, a compiler can either perform instantiation-time patching of immediates or use indirect through the global table.  This has the same tradeoffs (dirtying pages vs. GOT-indirection) as with normal dynamic linking.

Mutable variables need to be supported for engines that compile asm.js to wasm anyway, so it' no big deal to support in the MVP and might actually be useful for a tool (e.g., asm.js2wasm) that needs to allocate memory without perturbing linear memory.  Importing/Exporting mutable global variables is an extra burden, though, and that is explicitly ruled out in the MVP if you look at JS.md.